### PR TITLE
conversations SDK

### PR DESCRIPTION
**Pull request by mshaban-msft**
continuing the work of @annatisch
original PR link [here](https://github.com/Azure/azure-sdk-for-python/pull/20651)

> notes:
latest swagger
- repo: [here](https://github.com/Azure/azure-rest-api-specs/tree/feature/cognitiveservices/language/specification/cognitiveservices/data-plane/Language)
- branch: feature/cognitiveservices/language
- path: /specification/cognitiveservices/data-plane/Language/
- file: preview/2021-07-15-preview/analyzeconversations.json

### CODE REVIEW

File: eng/tox/allowed_pylint_failures.py [hunk 1]
```diff
@@ -58,5 +58,6 @@
     "azure-messaging-nspkg",
     "azure-agrifood-farming",
     "azure-eventhub",
-    "azure-ai-language-questionanswering"
+    "azure-ai-language-questionanswering",
+    "azure-ai-language-conversations"
```
Comment (iscai-msft) [inline eng/tox/allowed_pylint_failures.py:62]: add a changelog with initial release info

File: sdk/cognitivelanguage/azure-ai-language-conversations/README.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
 [![Build Status](https://dev.azure.com/azure-sdk/public/_apis/build/status/azure-sdk-for-python.client?branchName=main)](https://dev.azure.com/azure-sdk/public/_build/latest?definitionId=46?branchName=main)
 
-# Azure Cognitive Language Services Conversations client library for Python
+# Azure Cognitive Language Services Conversational Language Understanding client library for Python
+Conversational Language Understanding, aka LUIS vNext and **CLU** for short, is a cloud-based conversational AI service that applies custom machine-learning intelligence to a user's conversational, natural language text to predict overall meaning, and pull out relevant, detailed information (namely intents and entities).
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-conversations/README.md:4]: need to add that 2.7 will be deprecated soon, and add 3.10 to setup.py classifiers, see this code https://github.com/Azure/azure-sdk-for-python/pull/20975/commits/02ea20cec01dd5ebe3ed3658efde5eb5a5710f13

File: sdk/cognitivelanguage/azure-ai-language-conversations/setup.py [hunk 3]
```diff
@@ -1,11 +1,11 @@
-# coding=utf-8
+#!/usr/bin/env python
 # --------------------------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # Code generated by Microsoft (R) AutoRest Code Generator.
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
-# coding: utf-8
+# coding=utf-8
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-conversations/setup.py]: remove this line

File: sdk/cognitivelanguage/azure-ai-language-conversations/tests/asynctestcase.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# coding: utf-8
+# coding=utf-8
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-conversations/tests/asynctestcase.py:1]: no need for these changes I think

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/sample_authentication.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# coding=utf-8
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-conversations/samples/sample_authentication.py:1]: make sure to add a samples/README.md

File: sdk/cognitivelanguage/azure-ai-language-conversations/README.md [hunk 6]
```diff
@@ -1,6 +1,15 @@
 [![Build Status](https://dev.azure.com/azure-sdk/public/_apis/build/status/azure-sdk-for-python.client?branchName=main)](https://dev.azure.com/azure-sdk/public/_build/latest?definitionId=46?branchName=main)
 
-# Azure Cognitive Language Services Conversations client library for Python
+# Azure Cognitive Language Services Conversational Language Understanding client library for Python
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-conversations/README.md:3]: I think we should keep it the same as before, this is too long

File: sdk/cognitivelanguage/azure-ai-language-conversations/README.md [hunk 7]
```diff
@@ -22,21 +31,44 @@ pip install azure-ai-language-conversations
...
 
 
 #### Create ConversationAnalysisClient
+Once you've determined your **endpoint** and **API key** you can instantiate a `ConversationAnalysisClient`:
+
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.language.conversations import ConversationAnalysisClient
+
+endpoint = "https://{myaccount}.api.cognitive.microsoft.com"
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-conversations/README.md:57]: Update to be of format `https://<"my-account-name">.cognitiveservices.azure.com`

File: sdk/cognitivelanguage/azure-ai-language-conversations/tests/asynctestcase.py (hunk 4 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/tests/asynctestcase.py:1]: it's working! let's keep it as it is

File: sdk/cognitivelanguage/azure-ai-language-conversations/setup.py [hunk 8]
```diff
@@ -19,7 +18,7 @@
 # prerequisite: setuptools
 # http://pypi.python.org/pypi/setuptools
 
-REQUIRES = ["msrest>=0.6.21", "azure-core<2.0.0,>=1.16.0"]
+REQUIRES = ["msrest>=0.6.21", "azure-core<2.0.0,>=1.19.0"]
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-conversations/setup.py:21]: the setup.py should be updated to look like [this](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/textanalytics/azure-ai-textanalytics/setup.py). This is more of a generated basic setup.py, but we want to add things like the readme to the package

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/README.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
+---
+page_type: sample
+languages:
+  - python
+products:
+  - azure
+  - azure-cognitive-services
+  - azure-ai-language-conversations
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/samples/README.md:8]: The product slug indicates where the samples should be published. To me, azure-language-understanding seems correct. Full list here: https://review.docs.microsoft.com/en-us/help/contribute/metadata-taxonomies?branch=master#product

```suggestion
  - azure-language-understanding
```

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/README.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+urlFragment: conversationslanhguageunderstanding-samples
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/samples/README.md:9]: typo

```suggestion
urlFragment: conversationslanguageunderstanding-samples
```

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/async/sample_authentication_async.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
+# coding=utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_authentication_asyncasync.py
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/samples/async/sample_authentication_async.py:9]: and in below for USAGE

```suggestion
FILE: sample_authentication_async.py
```

File: sdk/cognitivelanguage/azure-ai-language-conversations/setup.py [hunk 12]
```diff
@@ -1,39 +1,91 @@
...
+        "Development Status :: 4 - Beta",
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
+        'License :: OSI Approved :: MIT License',
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/setup.py:72]:
```suggestion
        'Programming Language :: Python :: 3.9',
        'Programming Language :: Python :: 3.10',
        'License :: OSI Approved :: MIT License',
```

File: sdk/cognitivelanguage/azure-ai-language-conversations/setup.py [hunk 13]
```diff
@@ -1,39 +1,91 @@
...
+    install_requires=[
+        "azure-core<2.0.0,>=1.19.0",
+        "msrest>=0.6.21",
+        'azure-common~=1.1',
+        'six>=1.11.0',
+    ],
+    extras_require={
+        ":python_version<'3.0'": ['azure-ai-nspkg'],
+        ":python_version<'3.5'": ['typing'],
+    }
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/setup.py:90]: add

    project_urls={
        'Bug Reports': 'https://github.com/Azure/azure-sdk-for-python/issues',
        'Source': 'https://github.com/Azure/azure-sdk-python',
    }

**Review by kristapratico — COMMENTED** [review]
mostly nits and some feedback on the setup.py.

**Review by kristapratico — APPROVED** [review]

**Pull request merged**